### PR TITLE
update electra specs to `v1.5.0-alpha.6`

### PR DESCRIPTION
**Pull request by pk910**
Updates electra types for latest `v1.5.0-alpha.6` spec release:
* Reverted `electra.ExecutionPayload` & `electra.ExecutionPayloadHeader`
* Created new `electra.ExecutionRequests` type & added to `electra.BeaconBlockBody`
* merged latest master branch to fix unrelated linter errors

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
Don't you need the `RequestHash` in the ExecutionPayloadHeader?

**Comment by pk910** [comment]
All changes to the `ExecutionPayloadHeader` have been reverted,  it's no longer in the list of modified containers:
https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#modified-containers

so I guess, no? :D

**Comment by mcdee** [comment]
@pk910 when this is ready please tag me as a reviewer and I'll merge it.  Thanks.

**Comment by pk910** [comment]
Oh yea, it's ready @mcdee :)
I unfortunately can't assign a reviewer ;)

File: api/v1/forkchoice.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func (d *ForkChoiceNodeValidity) UnmarshalJSON(input []byte) error {
 
 // String returns a string representation of the ForkChoiceNodeValidity.
 func (d ForkChoiceNodeValidity) String() string {
-	if int(d) >= len(ForkChoiceNodeValidityStrings) {
+	if uint64(d) >= uint64(len(ForkChoiceNodeValidityStrings)) {
```
Comment (jtraglia) [inline api/v1/forkchoice.go:150]: What's the rationale for this change? I see this in a few other spots too.

File: http/spec.go [hunk 2]
```diff
@@ -121,8 +121,8 @@ func (s *Service) Spec(ctx context.Context,
 
 		// Handle durations.
 		if strings.HasPrefix(k, "SECONDS_PER_") || k == "GENESIS_DELAY" {
-			intVal, err := strconv.ParseUint(v, 10, 64)
-			if err == nil && intVal != 0 {
+			intVal, err := strconv.ParseInt(v, 10, 64)
+			if err == nil && intVal >= 0 {
```
Comment (jtraglia) [inline http/spec.go:125]: This allows a zero value now. Is that intended?

File: spec/electra/generate.go [hunk 3]
```diff
@@ -15,6 +15,6 @@ package electra
...
+//go:generate sszgen --suffix=ssz --path . --include ../phase0,../altair,../bellatrix,../capella,../deneb --objs AggregateAndProof,Attestation,AttesterSlashing,BeaconBlockBody,BeaconBlock,BeaconState,Consolidation,DepositReceipt,ExecutionLayerWithdrawalRequest,ExecutionRequests,PendingBalanceDeposit,PendingConsolidation,PendingPartialWithdrawal,SignedAggregateAndProof,SignedBeaconBlock
+//go:generate goimports -w aggregateandproof_ssz.go attestation_ssz.go attesterslashing_ssz.go beaconblockbody_ssz.go beaconblock_ssz.go beaconstate_ssz.go consolidation_ssz.go depositreceipt_ssz.go executionlayerwithdrawalrequest_ssz.go executionpayload_ssz.go executionpayloadheader_ssz.go pendingbalancedeposit_ssz.go pendingconsolidation_ssz.go pendingpartialwithdrawal_ssz.go signedaggregateandproof_ssz.go signedbeaconblock_ssz.go
```
Comment (jtraglia) [inline spec/electra/generate.go:20]: We should add `executionrequests_ssz.go` to the `rm -rf` and `goimports` lines too.

**Review by jtraglia — COMMENTED** [review]
LGTM! A couple questions & one suggestion.

**Comment by jtraglia** [comment]
@pk910, I've made a similar PR for `go-builder-client` which MEV software needs:

* https://github.com/attestantio/go-builder-client/pull/31

I noticed this when integrating your branch (this PR) into `flashbots/go-boost-utils`:

image

File: api/v1/forkchoice.go (hunk 1 above)
Comment (pk910) [reply api/v1/forkchoice.go:150]: This change is coming from master (50179b8bd4d75c956248c8af698373fac1ddb6c9).
I've merged in master due to unrelated [linter issues](https://github.com/attestantio/go-eth2-client/actions/runs/10999190743/job/30538823000) that were already fixed in master.
sorry for the messed up diff.

File: http/spec.go (hunk 2 above)
Comment (pk910) [reply http/spec.go:125]: Same as above. coming from master (https://github.com/attestantio/go-eth2-client/commit/50179b8bd4d75c956248c8af698373fac1ddb6c9).

File: api/v1/forkchoice.go (hunk 1 above)
Comment (jtraglia) [reply api/v1/forkchoice.go:150]: Ah okay, that makes sense. No worries!

**Comment by pk910** [comment]
> @pk910, I've made a similar PR for `go-builder-client` which MEV software needs:
> 
> * [Make necessary changes for alpha.6 go-builder-client#31](https://github.com/attestantio/go-builder-client/pull/31)
> 
> I noticed this when integrating your branch (this PR) into `flashbots/go-boost-utils`:
> 
> image

Yea, this looks right.  The `electra.ExecutionPayloadHeader` & `electra.ExecutionPayload`types have been removed as these are unchanged compared to deneb.
So the downstream projects should use `deneb.ExecutionPayloadHeader` & `deneb.ExecutionPayload` instead.
The error seems to be coming from the go-builder-client which is still referring to these types?

**Comment by jtraglia** [comment]
Oh sorry for the confusion, I saw the errors, then realized that `go-builder-client` needed to be updated too.

Everything is sorted out now!

File: spec/electra/generate.go (hunk 3 above)
Comment (pk910) [reply spec/electra/generate.go]: found a few more missing or old files - fixed now
